apps/settings/js/wifi.js

```diff
@@ -283,8 +283,17 @@ navigator.mozL10n.once(function wifiSettings() {
     // Show connection status
     icon.classList.add('wifi-signal');
     if (WifiHelper.isConnected(network)) {
-      localize(small, 'shortStatus-connected');
-      icon.classList.add('connected');
+      var conStatus = WifiHelper.getConnectionStatus(network);
+      if (conStatus == 'connected') {
+        localize(small, 'shortStatus-connected');
+        icon.classList.add('connected');
+      }
+      else if (conStatus == 'connecting') {
+        localize(small, 'shortStatus-connecting');
+      }
+      else if (conStatus == 'associated') {
+        localize(small, 'shortStatus-associated');
+      }
```

Comment:
is there any other chances that we would get other states here ? or only these three states would happen in this place ? (Just a question, if these are all possible situations, then please ignore this comment ;) )